docs/aws/audit/snsmonitoring/rules/is_topic_cmk_encrypted.mdx

```diff
@@ -23,6 +23,202 @@ HIPAA, HITRUST, NISTCSF, PCIDSS
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Simple Notification Service (SNS) dashboard by selecting "Services" from the top menu, then typing "SNS" in the search box and selecting "Simple Notification Service".
+3. In the SNS dashboard, select "Topics" from the left-hand navigation pane. This will display a list of all the SNS topics available within your current AWS region.
+4. For each topic in the list, check the "KMS master key" column. If the column displays "Default", it means that the SNS topic is not encrypted with a customer-managed KMS key, indicating a potential misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the SNS topics in your AWS account. You can do this by using the following AWS CLI command:
+
+   ```
+   aws sns list-topics
+   ```
+   This command will return a list of all SNS topics in your AWS account.
+
+2. Once you have the list of SNS topics, you can check the encryption status of each topic. You can do this by using the following AWS CLI command:
+
+   ```
+   aws sns get-topic-attributes --topic-arn <TopicARN>
+   ```
+   Replace `<TopicARN>` with the ARN of the SNS topic you want to check. This command will return the attributes of the specified SNS topic.
+
+3. From the returned attributes, check the `KmsMasterKeyId` attribute. If this attribute is not present or is set to `alias/aws/sns`, then the SNS topic is not encrypted with a KMS CMK.
+
+4. Repeat steps 2 and 3 for each SNS topic in your AWS account.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including SNS.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Create a client for SNS and list all the SNS topics.
+
+```python
+sns = session.client('sns')
+response = sns.list_topics()
+```
+
+4. Iterate over the list of topics and check if they are encrypted using KMS CMKs. If the 'KmsMasterKeyId' attribute is not present or not set, then the SNS topic is not encrypted using KMS CMKs.
+
+```python
+for topic in response['Topics']:
+    topic_attributes = sns.get_topic_attributes(TopicArn=topic['TopicArn'])
+    if 'KmsMasterKeyId' not in topic_attributes['Attributes']:
+        print(f"SNS Topic {topic['TopicArn']} is not encrypted using KMS CMKs.")
+```
+
+This script will print out the ARNs of all SNS topics that are not encrypted using KMS CMKs.
+</Accordion>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>